rest_framework/routers.py

```diff
@@ -277,10 +285,24 @@ def get_api_root_view(self):
         for prefix, viewset, basename in self.registry:
             api_root_dict[prefix] = list_name.format(basename=basename)
 
+        view_renderers = list(api_settings.DEFAULT_RENDERER_CLASSES)
+
+        if schema_urls and self.schema_title:
+            assert coreapi, '`coreapi` must be installed for schema support.'
+            view_renderers += [renderers.CoreJSONRenderer]
+            schema_generator = SchemaGenerator(patterns=schema_urls)
+
         class APIRoot(views.APIView):
             _ignore_model_permissions = True
+            renderer_classes = view_renderers
 
             def get(self, request, *args, **kwargs):
+                if request.accepted_renderer.format == 'corejson':
```

Comment:
The problem is less what to do with `CoreJSONRenderer` and more about making sure that we preserve the behavior of that view for anything that *isn't* a schema renderer.